Mathlib/Analysis/InnerProductSpace/JointEigenspace.lean

```diff
@@ -28,32 +31,50 @@ and a proof obligation that the basis vectors are eigenvectors.
 
 ## Tags
 
-self-adjoint operator, simultaneous eigenspaces, joint eigenspaces
+symmetric operator, simultaneous eigenspaces, joint eigenspaces
 
 -/
 
-variable {𝕜 E : Type*} [RCLike 𝕜]
-variable [NormedAddCommGroup E] [InnerProductSpace 𝕜 E]
-
 open Module.End
 
 namespace LinearMap
 
 namespace IsSymmetric
 
-section Pair
+variable {𝕜 E n m : Type*}
+
+section CommRing
 
+variable [CommRing 𝕜] [AddCommGroup E] [Module 𝕜 E]
 variable {α : 𝕜} {A B : E →ₗ[𝕜] E}
 
-/--If a pair of operators commute, then the eigenspaces of one are invariant under the other.-/
+/-- If a pair of operators commute, then the eigenspaces of one are invariant under the other. -/
 theorem eigenspace_invariant_of_commute
-    (hAB : A ∘ₗ B = B ∘ₗ A) (α : 𝕜) : ∀ v ∈ (eigenspace A α), (B v ∈ eigenspace A α) := by
+    (hAB : A ∘ₗ B = B ∘ₗ A) (α : 𝕜) : ∀ v ∈ eigenspace A α, B v ∈ eigenspace A α := by
   intro v hv
   rw [eigenspace, mem_ker, sub_apply, Module.algebraMap_end_apply, ← comp_apply A B v, hAB,
     comp_apply B A v, ← map_smul, ← map_sub, hv, map_zero] at *
 
-/--The simultaneous eigenspaces of a pair of commuting symmetric operators form an
-`OrthogonalFamily`.-/
+/-- The indexed infimum of eigenspaces of a commuting family of linear operators is
+invariant under each operator. -/
+theorem iInf_eigenspace_invariant_of_commute {T : n → E →ₗ[𝕜] E}
+    (hC : ∀ i j, T i ∘ₗ T j = T j ∘ₗ T i) (i : n) (γ : {x // x ≠ i} → 𝕜) ⦃v : E⦄
+    (hv : v ∈ ⨅ j, eigenspace (Subtype.restrict (· ≠ i) T j) (γ j)) :
+    T i v ∈ ⨅ j, eigenspace (Subtype.restrict (· ≠ i) T j) (γ j) := by
+  simp only [Submodule.mem_iInf] at hv ⊢
+  exact fun j ↦ eigenspace_invariant_of_commute (hC j i) (γ j) v (hv j)
```

Comment:
One final remark, it might be worth considering changing the type of `μ` to `μ : n → R` even though its value at `i` is irrelevant. My _guess_ is that most users will have a `μ` of this type and those that don't should still be able to use this lemma by cooking up a junk value. Furthermore, it results in a much simpler statement.

I'm not 100% sure I'm right here so I provide the following merely as food for thought:
```
example
    {T : n → Module.End R E} (hC : Pairwise fun i j ↦ Commute (T i) (T j))
    (i : n) (μ : n → R) (k : ℕ) :
    MapsTo (T i)
      (⨅ j ≠ i, genEigenspace (T j) (μ j) k)
      (⨅ j ≠ i, genEigenspace (T j) (μ j) k) := by
  intro v hv
  simp only [Set.iInf_eq_iInter, Set.mem_iInter, SetLike.mem_coe, Subtype.forall] at hv ⊢
  intro j hj
  exact mapsTo_genEigenspace_of_comm (hC hj) (μ j) k (hv j hj)
```
In fact come to think of it, this whole lemma is too specialised: it's irrelevant that `T i` is part of the family and the commtuation condition is too strong. This seems like the right generalisation:
```lean
example
    {f : n → Module.End R E} {g : Module.End R E} (hC : ∀ j, Commute (f j) g)
    (μ : n → R) (k : ℕ) :
    MapsTo g
      (⨅ j, genEigenspace (f j) (μ j) k)
      (⨅ j, genEigenspace (f j) (μ j) k) := by
  intro v hv
  simp only [Set.iInf_eq_iInter, Set.mem_iInter, SetLike.mem_coe, Subtype.forall] at hv ⊢
  intro j
  exact mapsTo_genEigenspace_of_comm (hC j) (μ j) k (hv j)
```
OK I really am out of time now!

Reply:
@ocfnash : Thanks for all of these suggestions! I will try to incorporate them carefully. (It may take a bit, as I'm still relatively inexperienced.)